arangod/Cluster/ClusterInfo.cpp

```diff
@@ -296,6 +299,105 @@ class CollectionWatcher
   std::atomic<bool> _present;
 };
 
+constexpr frozen::unordered_map<std::string_view, ServerHealth, 3>
+    kHealthStatusMap = {
+        {consensus::Supervision::HEALTH_STATUS_BAD, ServerHealth::kBad},
+        {consensus::Supervision::HEALTH_STATUS_FAILED, ServerHealth::kFailed},
+        {consensus::Supervision::HEALTH_STATUS_GOOD, ServerHealth::kGood},
+        {consensus::Supervision::HEALTH_STATUS_UNCLEAR,
+         ServerHealth::kUnclear}};
+
+[[nodiscard]] ServersKnown parseServersKnown(
+    VPackSlice const serversKnownSlice, VPackSlice const supervisionHealth,
+    containers::FlatHashSet<ServerID> const& serverIds) {
+  ServersKnown res;
+  TRI_ASSERT(serversKnownSlice.isNone() || serversKnownSlice.isObject());
+  LOG_TOPIC("91da8", TRACE, Logger::CLUSTER)
+      << "Supervision health is:" << supervisionHealth.toString();
+  if (serversKnownSlice.isObject()) {
+    for (auto const it : VPackObjectIterator(serversKnownSlice)) {
+      auto status = ServerHealth::kUnclear;
+      std::string serverId = it.key.copyString();
+      if (ADB_LIKELY(supervisionHealth.isObject())) {
+        try {
+          auto decoded = kHealthStatusMap.find(
+              supervisionHealth.get(serverId).get("Status").stringView());
+          if (decoded != kHealthStatusMap.end()) {
+            status = decoded->second;
+          }
+        } catch (velocypack::Exception const& ex) {
+          LOG_TOPIC("d783d", WARN, Logger::CLUSTER)
+              << "Invalid health data for server " << serverId
+              << " Error: " << ex.what()
+              << " Data: " << supervisionHealth.toString();
```

Comment:
Should this function go on after having caught the exception, or do we want it to exit it prematurely? 
I don't see what else it could, but just wondering.

Reply:
In this case it leaves state of server as `ServerHealth::kUnclear`.
I want to make changes here with as fewer side effects as possible. So function will go on.  As we actually got reboot id for server and all machinery could work as before. In general - having here something unexpected means we have something strange in the Agency. Maybe an assertion? WDYT?